bundles/specmate-cdo-server/src/com/specmate/cdoserver/internal/SpecmateCDOServer.java

```diff
@@ -0,0 +1,150 @@
+package com.specmate.cdoserver.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.emf.cdo.server.CDOServerUtil;
+import org.eclipse.emf.cdo.server.IRepository;
+import org.eclipse.emf.cdo.server.net4j.CDONet4jServerUtil;
+import org.eclipse.net4j.Net4jUtil;
+import org.eclipse.net4j.acceptor.IAcceptor;
+import org.eclipse.net4j.tcp.TCPUtil;
+import org.eclipse.net4j.util.container.IPluginContainer;
+import org.eclipse.net4j.util.lifecycle.LifecycleUtil;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+import com.specmate.cdoserver.ICDOServer;
+import com.specmate.cdoserver.config.SpecmateCDOServerConfig;
+import com.specmate.common.SpecmateException;
+import com.specmate.common.SpecmateValidationException;
+import com.specmate.dbprovider.api.DBConfigChangedCallback;
+import com.specmate.dbprovider.api.IDBProvider;
+import com.specmate.migration.api.IMigratorService;
+
+@Component(immediate = true, configurationPid = SpecmateCDOServerConfig.PID, configurationPolicy = ConfigurationPolicy.REQUIRE)
+public class SpecmateCDOServer implements DBConfigChangedCallback, ICDOServer {
+
+	/** The configured tcp port */
+	private int port;
+
+	/** The tcp acceptor */
+	private IAcceptor acceptorTCP;
+
+	/** The CDO repository */
+	private IRepository repository;
+
+	/** The CDO container */
+	private IPluginContainer container;
+
+	/** Reference to the DB provider service */
+	private IDBProvider dbProviderService;
+
+	/** Reference to the migration service */
+	private IMigratorService migrationService;
+
+	@Activate
+	public void activate(Map<String, Object> properties) throws SpecmateValidationException, SpecmateException {
+		readConfig(properties);
+		start();
+	}
+
+	@Deactivate
+	public void deactivate() {
+		shutdown();
+	}
+
+	/**
+	 * Reads the config properties
+	 *
+	 * @param properties
+	 * @throws SpecmateValidationException
+	 *             if the configuration is invalid
+	 */
+	private void readConfig(Map<String, Object> properties) throws SpecmateValidationException {
+		String portString = (String) properties.get(SpecmateCDOServerConfig.KEY_SERVER_PORT);
+		try {

```

Comment:
Read here also the repository string, so that getRepository() can be removed form the IDBProvider and implementing classes.

Reply:
done